client/utils/use-confirm-navigation.js

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useCallback, useRef } from '@wordpress/element';
+
+/**
+ * Hook for displaying an optional confirmation message.
+ *
+ * Usage:
+ * - const callback = useConfirmNavigation( true );
+ *   useEffect( callback , [ callback, otherDependency ] );
+ *
+ * @param {boolean} displayPrompt Wether we should prompt the message or not
```

Comment:
Minor typo: "Wether" should be "Whether".

Reply:
done 363597a2